package br.com.tonypool.atendimento.model;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@Table(name = "atendimento")
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Atendimento {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "idatendimento")
	private Integer idAtendimento;

	@Temporal(TemporalType.TIMESTAMP)
	@Column(name = "datahora", nullable = false)
	private Date dataHora;

	@Column(name = "observacoes", length = 500, nullable = false)
	private String observacoes;

	@Column(name = "idcliente", nullable = false)
	private Integer idCliente;

	@ManyToOne
	@JoinColumn(name = "idservico", nullable = false)
	private Servico servico;
	
	@ManyToOne
	@JoinColumn(name = "idprofissional", nullable = false)
	private Profissional profissional;
	
	@Column(name = "status", length = 20, nullable = false)
	private String status = "ABERTO";
}



